src/adapters/kafka/index.ts

```diff
@@ -0,0 +1,96 @@
+import fs from 'fs'
+import { Kafka } from 'kafkajs'
+import Adapter from '../../lib/adapter.js'
+import GleeMessage from '../../lib/message.js'
+
+class KafkaAdapter extends Adapter {
+  private kafka: Kafka
+  private firstConnect: boolean = true
+  name(): string {
+    return 'Kafka adapter'
+  }
+
+  async connect() {
+    const kafkaOptions = await this.resolveProtocolConfig('kafka')
+    const securityRequirements = (this.AsyncAPIServer.security() || []).map(
+      (sec) => {
+        const secName = Object.keys(sec.json())[0]
+        return this.parsedAsyncAPI.components().securityScheme(secName)
+      }
+    )
+    const userAndPasswordSecurityReq = securityRequirements.find(
+      (sec) => sec.type() === 'userPassword'
+    )
+    const brokerUrl = new URL(this.AsyncAPIServer.url())
+    this.kafka = new Kafka({
+      clientId: 'glee-app',  // clientID: hardcoded need to change afterwards 
+      brokers: [brokerUrl.host],
+      ssl: {
+        rejectUnauthorized: true,
+        key: fs.readFileSync('client.key'),
+        cert: undefined
+      },
+      sasl: {
+        mechanism: 'scram-sha-512',
+        username: process.env.KAFKA_USERNAME
+        ? kafkaOptions?.authentication?.userPassword?.username
+        : undefined,
+        password: process.env.KAFKA_PASSWORD
+        ? kafkaOptions?.authentication?.userPassword?.username
+        : undefined,
```

Comment:
You don't have to check environment variables now. If the user wants to use environment variables, they can do it in the config file. The only thing you have to do now is to check the config file values:
```suggestion
        username: kafkaOptions?.authentication?.userPassword?.username,
        password: kafkaOptions?.authentication?.userPassword?.username,
```